### PR TITLE
Added maven dependency for Bouncy Castle to server

### PR DESCRIPTION
**Pull request by lutovich**
It is used by Netty for self-signed certificate generation. It is a fallback path if proprietary package `sun.security.x509` provided by OpenJDK is not in the classpath. In particular, IBM JDK does not include this package.

### CODE REVIEW

**Comment by lutovich** [comment]
@jakewins could you please review this PR?

File: pom.xml [hunk 1]
```diff
@@ -27,6 +27,7 @@
     <lucene.version>3.6.2</lucene.version>
     <powermock.version>1.5.5</powermock.version>
     <logback-classic.version>1.1.2</logback-classic.version>
+    <bouncycastle.version>1.52</bouncycastle.version>
```
Comment (jotomo) [inline pom.xml:30]: Current version is the recently released 1.53 , which has some compatibility fixes that might be of interest to the server:
`The provider has also been updated to reflect changes in JDK 1.8 which broke X509Certificate.hashCode() and X509Certificate.verify(PublicKey, Provider)` 
from: http://bouncycastle.org/latest_releases.html

File: community/bolt/transport-socket/pom.xml [hunk 2]
```diff
@@ -50,6 +50,15 @@
     </dependency>
 
     <dependency>
+      <groupId>org.bouncycastle</groupId>
+      <artifactId>bcprov-jdk15on</artifactId>
+    </dependency>
+    <dependency>
+      <groupId>org.bouncycastle</groupId>
+      <artifactId>bcpkix-jdk15on</artifactId>
```
Comment (jakewins) [inline community/bolt/transport-socket/pom.xml]: Do we need to add it to Bolt in 2.3? Since we dislodged Bolt from the deps, we shouldn't need it right? Suggest we do a separate PR for adding it to Bolt in 3.0, since the modules have changed there. (in fact, I've got a card to remove Bolt entirely from the 2.3 branch now that 2.3.0 is out).

**Comment by jakewins** [comment]
I like @jotomo's point - if there's a Java 8 fix in the next version, seems like a good idea to bump? Since this only goes in the server, bumping the dep version shouldn't affect anyones embedded dependency trees. 

Other than that and my little question re: Bolt, LGTM, happy to merge once those two things are decided on

File: pom.xml (hunk 1 above)
Comment (lutovich) [inline pom.xml:30]: Release 1.53 indeed worth upgrading to but probably in a separate PR. Here I've just extracted existing version as a parameter.

File: community/bolt/transport-socket/pom.xml (hunk 2 above)
Comment (lutovich) [inline community/bolt/transport-socket/pom.xml]: Bolt tests are still executed during the build. And without bouncycastle some tests fail with because neither `sun.security.x509` nor `org.bouncycastle.cert` are in the classpath.
If Bolt is going to be removed than it is better to wait with this PR and change only server pom.

Comment (jakewins) [inline community/bolt/transport-socket/pom.xml]: Hmm. I'll write a PR right now then to remove Bolt.

**Comment by jakewins** [comment]
Bolt removal PR: https://github.com/neo4j/neo4j/pull/5822

**Comment by lutovich** [comment]
@jakewins Bolt removal merged, made this PR to only add bouncycastle to the server.

**Comment by tinwelint** [comment]
@lutovich @jakewins can we get this in now to help builds be happier?

**Comment by lutovich** [comment]
@tinwelint @jakewins this PR is ready to be merged.

File: pom.xml (hunk 1 above)
Comment (jotomo) [inline pom.xml:30]: @lutovich You still have this (PR for 1.53) on your radar or a card? :-)
